Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/PressableButtons/PressableButton.cs

```diff
@@ -218,69 +319,78 @@ void IMixedRealityTouchHandler.OnTouchCompleted(HandTrackingInputEventData event
             if (touchPoints.ContainsKey(eventData.Controller))
             {
                 touchPoints.Remove(eventData.Controller);
-
+                IsTouching = (touchPoints.Count > 0);
                 eventData.Use();
             }
         }
 
         #endregion OnTouch
 
-        #region private Methods
+        #region public transform utils
 
-        private void FindOrCreatePathMarkers()
+        /// <summary>
+        /// Returns world space position along the push direction for the given local distance
+        /// </summary>
+        /// <param name="localDistance"></param>
+        /// <returns></returns>
+        /// 
+        public Vector3 GetWorldPositionAlongPushDirection(float localDistance)
```

Comment:
Do these need to be public? They seem like they'd only be interesting within the class itself for figuring out where to position itself.

Reply:
i need them in the inspector class as well

Reply:
> This looks really promising! I did have some questions in the code, and also I foung a few bugs.
> 
> First, the blue push buttons do not move correctly -- they move in Y space when I pulled the change.
> 
> ![image](https://user-images.githubusercontent.com/168492/58219631-76310580-7cc0-11e9-897e-efce88f56210.png)
> 
> Also, to demonstrate that buttons can work at multiple scales now I added two buttons in PressableButtonExample scene with two scales. I think it would be good to add that so that people can try it out.
> 
> When adding that, I noticed another bug: when I scaled my HolographicButton to be size 0.1, the planes were still rendering quite big:
> 
> ![image](https://user-images.githubusercontent.com/168492/58219909-77166700-7cc1-11e9-9e74-6feefbf8b52a.png)
> 
> The buttons are also not actually pressing. Here is the package for you to test:
> [pressablebuttonexample.zip](https://github.com/microsoft/MixedRealityToolkit-Unity/files/3209978/pressablebuttonexample.zip)

thanks Julia!
I'll look into the push issues right away.

The distance planes only scale in push direction - the scale wouldn't have any effect on the width / height of the planes. I can look into adding that.

Reply:
Ah I see the problem with the rotation that gets lost - fixing that now

though i can't see the buttons that don't push in your example scene - for me the scene looks identical to the pressable button example scene that is in the demos and the buttons work fine for me.

Reply:
so i couldn't repro a problem with your scene but i found another problem that causes plane distances to be applied incorrectly and i assume that's the problem you were running into - preparing fix now

Reply:
just realized I wasn't seeing your scene but my local button demo scene. for some reason i can't import your scene?!

Reply:
Hm, that's weird. All I did was import HolographicButton prefab (which is quite large), then add a pressable button component to it. Then I scaled that to be 0.1 scale. That should repro the issue.

Reply:
i copied one of the pressable buttons with back plate from the example and scaled it to 0.1  - i can press it though it's very difficult bc it's so hard to catch xD. i will use a scale of 0.3 in the example scene

(tested with the new changes I pushed - rotations and debug visualisation offsets should be fixed by this)

Reply:
![image](https://user-images.githubusercontent.com/36998103/58283015-34c35780-7d9f-11e9-8bce-f2920a276a09.png)

Reply:
just merged in latest changes and the button example scene has been removed - i guess our button examples are part of the hand interaction scene - not sure where to put the buttons there to not make it look messy - will talk to yoon but would be great if we can still get this change in for the current release even without the example